apps/docs-snippets/src/guide/provider/querying-the-chain.test.ts

```diff
@@ -141,4 +142,101 @@ describe('querying the chain', () => {
     expect(message).toBeDefined();
     expect(message?.nonce).toEqual(nonce);
   });
+
+  it('can getMessage', async () => {
+    // #region Message-getMessages
+    // #import { TestMessage, launchTestNode };
+
+    // Creates a test message with an amount of 100
+    const testMessage = new TestMessage({ amount: 100 });
+
+    // Launches a test node with the test message configured
+    using launched = await launchTestNode({ walletsConfig: { messages: [testMessage] } });
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    // Retrieves messages from the wallet
+    const { messages } = await wallet.getMessages();
+    // #endregion Message-getMessages
+
+    expect(messages[0].nonce).toEqual(testMessage.nonce);
+  });
+
+  it('can getMessageProof with blockId', async () => {
+    // #region Message-getMessageProof-blockId
+    // #import { launchTestNode, TransactionResultMessageOutReceipt };
+
+    // Launches a test node
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [sender, recipient],
+      provider,
+    } = launched;
+
+    // Performs a withdrawal transaction from sender to recipient, thus generating a message
+    const withdrawTx = await sender.withdrawToBaseLayer(recipient.address.toB256(), 100);
+    const result = await withdrawTx.waitForResult();
+
+    // Emulate the next block creation by performing another tx
+    const transferTx = await sender.transfer(
+      recipient.address.toB256(),
+      100,
+      provider.getBaseAssetId()
+    );
+    const nextBlock = await transferTx.waitForResult();
+
+    // Retrieves the `nonce` via message out receipt from the initial transaction result
+    const { nonce } = result.receipts[0] as TransactionResultMessageOutReceipt;
+
+    // Retrieves the message proof for the transaction ID and nonce using the next block Id
+    const messageProof = await provider.getMessageProof(
+      result.gqlTransaction.id,
+      nonce,
+      nextBlock.blockId
+    );
+    // #endregion Message-getMessageProof-blockId
+
+    expect(messageProof?.amount.toNumber()).toEqual(100);
+    expect(messageProof?.sender.toHexString()).toEqual(result.id);
+  });
+
+  it('can getMessageProof with blockHeight', async () => {
+    // #region Message-getMessageProof-blockHeight
+    // #import { launchTestNode, TransactionResultMessageOutReceipt };
+
+    // Launches a test node
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [sender, recipient],
+      provider,
+    } = launched;
+
+    // Performs a withdrawal transaction from sender to recipient, thus generating a message
+    const withdrawTx = await sender.withdrawToBaseLayer(recipient.address.toB256(), 100);
+    const result = await withdrawTx.waitForResult();
+
+    // Emulate the next block creation by performing another tx
+    await sender.transfer(recipient.address.toB256(), 100, provider.getBaseAssetId());
```

Comment:
We have a problem here. This transfer is being made solely to create a new block. This dummy transfer transaction occurs for two reasons:

1. We need to wait for one confirmation block to obtain the message proof correctly; otherwise, `fuel-core` throws an error.

2. The `fuel-core` configuration for this test is set to spawn nodes instantly rather than at regular intervals. This configuration results in a new node spawning with every transaction submitted, this explains the existence of a dummy transfer transaction.

In real-world environments, such as the testnet and eventually the mainnet, nodes will spawn at regular intervals. Therefore, teaching users to submit a dummy transaction to wait for a new block is not practical.

Instead, the best approach would be to wait for one second, which is the default time for a new block to be generated within the Fuel ecosystem. By waiting for one second, we ensure that we have one confirmation block, allowing us to safely obtain the message proof.

```ts
  it('can getMessageProof with blockHeight', async () => {
    // Launches a test node
    using launched = await launchTestNode({
      nodeOptions: {
        args: ['--poa-instant', 'false', '--poa-interval-period', '1s'],
      },
    });

    const {
      wallets: [sender, recipient],
      provider,
    } = launched;

    // #region Message-getMessageProof-blockHeight
    // #import { TransactionResultMessageOutReceipt };

    // Performs a withdrawal transaction from sender to recipient, thus generating a message
    const withdrawTx = await sender.withdrawToBaseLayer(recipient.address.toB256(), 100);
    const result = await withdrawTx.waitForResult();

    // Waiting for a new block to be commited (1 confirmation block)
    await new Promise((resolve) => {
      setTimeout(() => {
        resolve(true);
      }, 1000);
    });

    // Retrieves the `nonce` via message out receipt from the initial transaction result
    const { nonce } = result.receipts[0] as TransactionResultMessageOutReceipt;

    // Retrives the latest block
    const latestBlock = await provider.getBlock('latest');

    // Retrieves the message proof for the transaction ID and nonce using the block height
    const messageProof = await provider.getMessageProof(
      result.gqlTransaction.id,
      nonce,
      undefined,
      latestBlock?.height
    );
    // #endregion Message-getMessageProof-blockHeight

    expect(messageProof?.amount.toNumber()).toEqual(100);
    expect(messageProof?.sender.toHexString()).toEqual(result.id);
  });
```

Reply:
Good catch, @Torres-ssf! I will fix this.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2716/commits/270b272856fbb4ec1a59b161e558a8ba2c228389